evil-commands.el

```diff
@@ -2021,6 +2023,58 @@ The return value is the yanked text."
   (interactive)
   (evil-paste-from-register ?.))
 
+(defun evil--get-common-indent-in-region (start end)
+  "Return the minimum indent (in columns) between START and END ignoring empty lines."
+  (let (indentation)
+    (save-excursion
+      (goto-char start)
+      (or (bolp) (forward-line 1))
+      (while (< (point) end)
+        (skip-chars-forward " \t")
+        (unless (eolp)
+          (setq indentation
+                (if indentation
```

Comment:
I think it would be cleaner to set `indentation` to 0 in the `let` form above and use `max` to get rid of this `if` branch. Then you can also remove the `or` in the return value and just return `indentation`.

Reply:
But would it still compute the same thing? The goal is to return the _minimum_ indentation, ignoring lines that contain only whitespace. Setting `indentation` to `most-positive-fixnum` and using `min` would work, but is it much more readable? I guess it depends on what should be returned if the region consists only of whitespace (currently `0` is returned).
```elisp
  (let ((indentation most-positive-fixnum))
    (save-excursion
      (goto-char start)
      (or (bolp) (forward-line 1))
      (while (< (point) end)
        (skip-chars-forward " \t")
        (unless (eolp)
          (setq indentation
                (min indentation (current-column))))
        (forward-line 1)))
    (if (= indentation most-positive-fixnum)
        0
      indentation))
```
(If you prefer this version, just say so then I'll gladly change it. :slightly_smiling_face:)

Reply:
Whoops your right, I wasn't paying attention to the loop. Sorry for the noise.

No it's not much more readable with the newer version, it just moves the if branch somewhere else. Although it is slightly more efficient as it moves the branch outside of the loop.